src/Symfony/Component/Messenger/Worker.php

```diff
@@ -113,6 +114,18 @@ private function handleMessage(Envelope $envelope, ReceiverInterface $receiver,
         try {
             $envelope = $this->bus->dispatch($envelope->with(new ReceivedStamp($transportName), new ConsumedByWorkerStamp()));
         } catch (\Throwable $throwable) {
+            if ($throwable instanceof StopWorkerException) {
```

Comment:
I already express my opinion about users who **can** use eventListerner to achieve the same behavior.

But I think, if we want providing an easy way to for end-user, we **should ** use eventListener here to avoid adding complexity in the worker